agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -143,6 +146,19 @@ func (state *DockerTaskEngineState) ContainerMapByArn(arn string) (map[string]*a
 	return ret, ok
 }
 
+// TaskByShortID retrieves the task of a given docker short container id
+func (state *DockerTaskEngineState) TaskByShortID(cid string) (*api.Task, bool) {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	for id := range state.idToTask {
+		if strings.HasPrefix(id, cid) {
```

Comment:
@samuelkarp I've been thinking that maybe this should be model like how the docker cli tools work allowing you to specify the the small number of characters if actually check for collisions and return an error if there is

something like this
```
docker logs c6
Error response from daemon: Multiple IDs found with provided prefix: c69a1491a2eab90f2153ea18b86ac43f24dbc4a3228b44363a67c21bddd3435a
````
WDYT?